k8s_e2e_test.go

```diff
@@ -197,6 +203,26 @@ var _ = Describe("CloudControllerManager", func() {
 					Expect(err).NotTo(HaveOccurred())
 				})
 			})
+			Context("a Metrics Server Helm Chart", func() {
+				BeforeEach(func() {
+					By("Initializing Helm")
+					helmInit()
+
+					By("Installing Metrics Server from Helm Chart")
+					installHelmChart(metricsServerName, "metrics-server/metrics-server")
+				})
+
+				AfterEach(func() {
+					By("Deleting Metrics Server Helm Chart")
+					deleteHelmChart(metricsServerName)
+				})
+
+				It("should successfully deploy Metrics Server helm chart and check it reports metrics, after 2 minutes", func() {
+					time.Sleep(2 * time.Minute)
+					_, err = f.Cluster.GetPodMetrics()
+					Expect(err).NotTo(HaveOccurred())
```

Comment:
Instead of a `fmt.Sleep()`, why not an `Eventually()` block where we retry (with a configurable timeout) until a success occurs? That way we can succeed faster if we get metrics before the 2m are up.

Reply:
That is a good point as things should be available after ~30 seconds based on my testing. I just pushed a change to add this.